code/Extensions/UsedOnTableExtension.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace SilverStripe\AssetAdmin\Extensions;
+
+
+use SilverStripe\Assets\Shortcodes\FileLink;
+use SilverStripe\Core\Extension;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * Class UsedOnTableExtension
+ *
+ * Hides File Links on the Used On tab when viewing files
+ */
+class UsedOnTableExtension extends Extension
+{
+    public function updateUsage(ArrayList &$usage, DataObject &$record) {
```

Comment:
Maybe use the more generic `SS_List` here. DataList tend to be more efficient because you can use SQL for your filtering. In the future, we may want to switch to a different list type. 

```suggestion
    public function updateUsage(SS_List &$usage, DataObject &$record) {
```